plumbing/format/packfile/decoder.go

```diff
@@ -62,6 +67,8 @@ type Decoder struct {
 
 	offsetToType map[int64]plumbing.ObjectType
 	decoderType  plumbing.ObjectType
+
+	cache *cache
```

Comment:
Maybe we could make cache public in its own package, since an object cache might be useful in other cases...